centreon-certified/bsm/bsm-events-apiv2.lua

```diff
@@ -0,0 +1,372 @@
+  end
+
+  self.sc_event.event.formated_event = {
+    hostname = self.sc_event.event.cache.host.name,
+    svc_desc = self.sc_event.event.cache.service.description,
+    state =  self.sc_event.event.state,
+    last_update = self.sc_event.event.last_update,
+    output =  string.match(self.sc_event.event.output, "^(.*)\n"),
+    service_severity = xml_service_severity,
+    url = xml_url,
+    source_host_id = self.sc_common:ifnil_or_empty(self.sc_event.event.host_id, 0),
+    source_svc_id = self.sc_common:ifnil_or_empty(self.sc_event.event.service_id, 0),
+    scheduled_downtime_depth = self.sc_common:ifnil_or_empty(self.sc_event.event.scheduled_downtime_depth, 0)
+  }
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:add method
+-- @param e An event
+--------------------------------------------------------------------------------
+
+function EventQueue:add()
+  -- store event in self.events lists
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+
+  self.sc_logger:debug("[EventQueue:add]: add event in queue category: " .. tostring(self.sc_params.params.reverse_category_mapping[category])
+    .. " element: " .. tostring(self.sc_params.params.reverse_element_mapping[category][element]))
+
+  self.sc_logger:debug("[EventQueue:add]: queue size before adding event: " .. tostring(#self.sc_flush.queues[category][element].events))
+  self.sc_flush.queues[category][element].events[#self.sc_flush.queues[category][element].events + 1] = self.sc_event.event.formated_event
+
+  self.sc_logger:info("[EventQueue:add]: queue size is now: " .. tostring(#self.sc_flush.queues[category][element].events)
+    .. "max is: " .. tostring(self.sc_params.params.max_buffer_size))
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} json encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} json encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+  if not payload then
+    payload = "<event_data>"
+    for index, xml_str in pairs(event) do
+      payload = payload .. "<" .. tostring(index) .. ">" .. tostring(self.sc_common:xml_escape(xml_str)) .. "</" .. tostring(index) .. ">"
+    end
+    payload = payload .. "</event_data>"
+
+  else
+    payload = payload .. "<event_data>"
+    for index, xml_str in pairs(event) do
+      payload = payload .. "<" .. tostring(index) .. ">" .. tostring(self.sc_common:xml_escape(xml_str)) .. "</" .. tostring(index) .. ">"
+    end
+    payload = payload .. "</event_data>"
+  end
+
+  return payload
+end
+
+function EventQueue:send_data(payload)
+  self.sc_logger:debug("[EventQueue:send_data]: Starting to send data")
+
```

Comment:
before the send _data_test condition, there is a whole new system to create the curl command.

in this case it would look like 

```lua
queue_metadata.headers = {
  "Content-Type: text/xml",
  "content-length: " .. string.len(payload)
}


self.sc_logger:log_curl_command(self.sc_params.params.http_server_url, queue_metadata, self.sc_params.params, payload)
```